src/osd/scrubber_common.h

```diff
@@ -415,8 +415,33 @@ struct ScrubPgIF {
 
   // --------------- debugging via the asok ------------------------------
 
-  virtual int asok_debug(std::string_view cmd,
-			 std::string param,
-			 Formatter* f,
-			 std::stringstream& ss) = 0;
+  virtual int asok_debug(
+      Formatter* f,
+      std::string_view prefix,
+      std::string_view cmd,
+      std::string_view param,
+      std::string_view val) = 0;
 };
+
+
+// compile-time hashing of strings for fast compare
+constexpr inline uint32_t cx_hash(std::string_view str )
+{
+    uint32_t hash = 0;
+    for (auto c : str)
+    {
+        hash += c;
+        hash += (hash << 10);
+        hash ^= (hash >> 6);
+    }
+
+    hash += (hash << 3);
+    hash ^= (hash >> 11);
+    hash += (hash << 15);
+
+    return hash;
+}
+
+constexpr unsigned operator ""_xhash(const char* s, size_t) {
+    return cx_hash(s);
+}
```

Comment:
IIUC, this will be used for debugging/testing only. Shouldn't we prefer readability over performance in this case?

Reply:
I intend to use this mechanism for all our long "if (cmd == "word1") { } else if (cmd == "word2") ...
(eg in OSD.cc and PrimaryLogPG.cc).

And I really do not think there is any clarity issue with
"commandstring"_xhash
as a way to signal the hash of the string.

BTW - that's not my invention... it's a pretty common way to avoid costly strings compare. But it was harder to implement before C++17.

Reply:
I would vote for coding it as simple and dumb as possible, disregarding the performance entirely -- I doubt it plays any role in asok.

Reply:
> BTW - that's not my invention... it's a pretty common way to avoid costly strings compare. But it was harder to implement before C++17.

Matan's point was the performance in this particular appliance is just irrelevant; only easiness-of-understanding matters. I agree there is no benefit in spending the complexity budget on the asok sutff – the cheaper (in terms of mental load), the better.